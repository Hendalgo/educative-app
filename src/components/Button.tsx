import React, {useState} from "react";
import { Pressable, StyleProp } from "react-native";
import { useTheme } from "@react-navigation/native";
import { CustomColors } from "src/styles/themes";
import styles from "../styles";
import { Text } from "react-native";


interface ButtonProps {
  title: string;
  onPress: () => void;
  color?: string;
  type?: 'primary' | 'secondary' | 'disable' | 'warning' | 'success' | 'info' | 'danger' | 'link' | 'text' | 'outline' | 'ghost' | 'dashed' | 'default' | 'text';
}

const Button = ({ title, onPress, type = 'default' }: ButtonProps): React.JSX.Element => {
  //@ts-ignore
  const { colors }: { colors: CustomColors } = useTheme();
  //const [stylesInputs, setStylesInputs] = useState<StyleProp<any>>({});
  const styleButton:StyleProp<any> = {}
  const styleText:StyleProp<any> = {}

  const styles = {
    disable: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
      styleText.color = colors.neutral000;
    },
    primary: (): void=>{
      styleButton.backgroundColor = colors.primary000;
      styleButton.borderColor = colors.primary000;
      styleText.color = colors.neutral000;
    },
    secondary: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    warning: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    success: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    info: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    danger: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    link: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    text: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    outline: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    ghost: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    dashed: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    },
    default: (): void=>{
      styleButton.backgroundColor = colors.primary1200;
      styleButton.borderColor = colors.primary1200;
    }
  }
  styles[type] && styles[type]();
  return (
    <Pressable
      style={[
        {
          ...styleButton,
          padding: 14,
          borderRadius: 8,
        },
      ]}
      onPress={onPress}
    >
      <Text
        style={{
          ...styleText,
          fontSize: 16,
          fontFamily: 'Poppins-Regular',
          textAlign: 'center',
          textAlignVertical: 'center',
        }}
      >
        {title}
      </Text>
    </Pressable>
  );
};

export default Button;