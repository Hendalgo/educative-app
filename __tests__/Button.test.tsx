import React from "react";
import Button from "@components/Button";
import { fireEvent, render } from "@testing-library/react-native";

const ButtonTest = ({typeB = 'default', onPress= ()=>{}}: {typeB?:any, onPress?:()=>void}): React.JSX.Element => {
  return (
    <Button title="Test" type={typeB} onPress={onPress} />
  );
}

describe("Test Button Componet", () => {
  it("Should render the button component", () => {
    const wrapper = render(<ButtonTest />);
    expect(wrapper).toMatchSnapshot();
  });
  it("Should render the button component with the title Test", () => {
    const wrapper = render(<ButtonTest />);
    expect(wrapper.getByText("Test")).toBeDefined();
  });
  it("render the button component with the type disable", () => {
    const wrapper = render(<ButtonTest typeB="disable" />);
    expect(wrapper.getByText("Test")).toBeDefined();
  });
  it("on press event should be called", () => {
    const handlePress = jest.fn();
    const {getByText} = render(<ButtonTest onPress={handlePress} />);
    fireEvent.press(getByText("Test"));
    expect(handlePress).toHaveBeenCalled();
  });
});