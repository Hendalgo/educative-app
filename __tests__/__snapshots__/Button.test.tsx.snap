// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Componet Should render the button component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": undefined,
        "borderColor": undefined,
        "borderRadius": 8,
        "padding": 14,
      },
    ]
  }
>
  <Text
    style={
      {
        "fontFamily": "Poppins-Regular",
        "fontSize": 16,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Test Button Componet render the button component with the type disable 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": undefined,
        "borderColor": undefined,
        "borderRadius": 8,
        "padding": 14,
      },
    ]
  }
>
  <Text
    style={
      {
        "color": undefined,
        "fontFamily": "Poppins-Regular",
        "fontSize": 16,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    Test
  </Text>
</View>
`;
